src/Nethermind/Nethermind.Api/Extensions/PluginLoader.cs

```diff
@@ -43,25 +43,26 @@ public void Load(ILogManager logManager)
         {
             ILogger logger = logManager.GetClassLogger();
             string baseDir = string.Empty.GetApplicationResourcePath();
-            string fullPluginsDir = _pluginsDirectory.GetApplicationResourcePath();;
-            if (!_fileSystem.Directory.Exists(fullPluginsDir))
+            string pluginAssembliesDir = _pluginsDirectory.GetApplicationResourcePath();
+            if (!_fileSystem.Directory.Exists(pluginAssembliesDir))
             {
-                if (logger.IsWarn) logger.Warn($"Plugins folder {fullPluginsDir} was not found. Skipping.");
+                if (logger.IsWarn) logger.Warn($"Plugin assemblies folder {pluginAssembliesDir} was not found. Skipping.");
             }
 
-            string[] pluginFiles = _fileSystem.Directory.GetFiles(fullPluginsDir).Where(p => p.EndsWith("dll")).ToArray();
-            if (pluginFiles.Length > 0)
+            string[] assemblies = _fileSystem.Directory.GetFiles(baseDir).Where(p => p.EndsWith("dll")).ToArray();
+            if (assemblies.Length > 0)
             {
-                if (logger.IsInfo) logger.Info($"Loading {pluginFiles.Length} plugins from {fullPluginsDir}");
+                if (logger.IsInfo) logger.Info($"Loading {assemblies.Length} assemblies from {baseDir}");
             }
 
-            foreach (string path in pluginFiles)
+            foreach (string path in assemblies)
             {
                 string pluginAssembly = _fileSystem.Path.GetFileNameWithoutExtension(path);
+
                 try
                 {
-                    if (logger.IsInfo) logger.Warn($"  Loading assembly {pluginAssembly}");
-                    string assemblyPath = _fileSystem.Path.Combine(fullPluginsDir, path);
+                    if (logger.IsInfo) logger.Warn($"Loading assembly {pluginAssembly}");
+                    string assemblyPath = _fileSystem.Path.Combine(pluginAssembliesDir, path);
```

Comment:
its more of fileName than a path here I think

Reply:
I think it is  a path

Reply:
if it was full pats you wouldn't combine it back with pluginAssembliesDir. "Returns the names of files that meet specified criteria." assemblyPath is correct naming

Reply:
rather assembly than fileName -> agree that path is bad naming